lld/ELF/InputSection.cpp

```diff
@@ -463,7 +463,10 @@ void InputSection::copyRelocations(uint8_t *buf,
 
       if (RelTy::IsRela)
         p->r_addend = sym.getVA(addend) - section->getOutputSection()->addr;
-      else if (config->relocatable && type != target.noneRel)
+      // See the comment in writeSections. For SHF_ALLOC sections relocated by
+      // REL relocations, utilize the non-REL/RELA loop.
+      else if (config->relocatable && (sec->flags & SHF_ALLOC) &&
+               type != target.noneRel)
         sec->addReloc({R_ABS, type, rel.offset, addend, &sym});
```

Comment:
Would it make sense to move the `(sec->flags & SHF_ALLOC)` check inside this if? That would ensure that we don't end up hitting later branches if we add some in the future?
```suggestion
        if ((sec->flags & SHF_ALLOC) == 0)
          sec->addReloc({R_ABS, type, rel.offset, addend, &sym});
```

Reply:
Thanks for looking. I'm puzzled. Do you mean to split the `else if ` into

```
else if (...) {
  if (!(sec->flags & SHF_ALLOC))
    sec->addReloc(...);
}
```

Reply:
Yes exactly that's what I meant. Should not have any functional difference to the current patch but might be safer for future refactoring. Feel free to ignore though :)

Reply:
I'll think I'll keep the current style :) I believe we will not add new `else if` to the existing `if .. else if`